symphony-bdk-http/symphony-bdk-http-jersey2/src/main/java/com/symphony/bdk/http/jersey2/ApiClientJersey2.java

```diff
@@ -212,7 +214,12 @@ private Response getResponse(Invocation.Builder invocationBuilder, String method
     } catch (ProcessingException e) {
       if (e.getCause() instanceof ConnectTimeoutException) {
         throw new ProcessingException(new SocketTimeoutException(e.getCause().getMessage()));
-      } else {
+      }
+      else if (e.getCause() instanceof NoHttpResponseException) {
+        // ensures that it will be caught later in the retry strategy
+        throw new ProcessingException(new SocketException(e.getCause().getMessage()));
```

Comment:
don't we want to chain the exception with the original one (passing e instead of just the message) so we don't loose the full stacktrace?

Reply:
We actually don't have access to the `NoHttpResponseException` on core side as it is a specific one from Jersey2. This way we ensure that it's properly caught on the RetryWithRecovery `side`

Reply:
`SocketException` constructor only accepts string